File: Python/sparse_matrix.py
def is_sparse(matrix):
    rows = len(matrix)
    cols = len(matrix[0])
    total = rows * cols
    count = 0

    for row in matrix:
        for val in row:
            if val == 0:
                count += 1

    return count > total / 2

# Input from user
r = int(input("Enter number of rows: "))
c = int(input("Enter number of columns: "))
matrix = []

print("Enter matrix row by row:")
for i in range(r):
    row = list(map(int, input().split()))
    matrix.append(row)

print("Matrix is Sparse:" if is_sparse(matrix) else "Matrix is Not Sparse")

File: C/sparse_matrix.c
#include<stdio.h>

int isSparse(int rows, int cols, int matrix[rows][cols]){
    int zero_count = 0;
    int size = rows*cols;

    for(int i = 0; i<rows; i++){
        for(int j = 0; j<cols; j++){
            if(matrix[i][j] == 0){
                zero_count++;
            }
        }
    }

    if (zero_count > size/2){
        return 1;
    }
    return 0;
}

int main(){
    int rows, cols;
    printf("Enter matrix dimensions: ");
    scanf("%d %d", &rows, &cols);
    
    int matrix[rows][cols];
    printf("Enter matrix elements:\n");
    for(int i = 0; i < rows; i++){
        for(int j = 0; j < cols; j++){
            scanf("%d", &matrix[i][j]);
        }
    }
    
    printf("\nOriginal Matrix:\n");
    for(int i = 0; i < rows; i++) {
        for(int j = 0; j < cols; j++){
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }
            
    if(isSparse(rows, cols, matrix)){
        printf("Matrix is sparse");
    }

    else{
        printf("Matrix is not sparse");
    }

    return 0;
}